### DISCUSSION TITLE
Stored Root, SubCA and dpvpn on swanctl

### DISCUSSION THREAD

**Discussion by dhodyrahmad** [strongswan/strongswan#495] [Q&A]
Hello, It my first time to setup the cert configuration to swanctl for my work. I've done read some PKI about storing cert on swanctl but still don't get it. How it supposed to store the Root, SubCA and  dpvpn cert on swanctl configuration? There's something I have to read first?

hope your feedback or suggestion about this problem. many thanks..

**Comment by Thermi** [comment]
Put them into the corresponding directories under swanctl/. The tool will figure out the rest.

Am July 5, 2021 7:29:16 AM UTC schrieb Dhody Rahmad Hidayat ***@***.***>:
>I've done read some PKI about storing cert on swanctl but still don't
>get it. How it supposed to store the Root, SubCA and  dpvpn cert on
>swanctl configuration?
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/strongswan/strongswan/discussions/495

-- 
Sent from mobile

**Reply by dhodyrahmad** [reply]
Hi, Thank you for your answer. Which directory to store for each certificate above? I'm sorry I really new to this. and how about the swanctl config file? which part I have to add to load the certificate? 

could you give some details or maybe some example to read and learn about, please?

**Reply by Thermi** [reply]
Hi,

See the wiki on the [directories](https://wiki.strongswan.org/projects/strongswan/wiki/SwanctlDirectory) and [examples for config files](https://wiki.strongswan.org/projects/strongswan/wiki/UsableExamples).
